internal/resources/gitrepository/spec/spec.go

```diff
@@ -0,0 +1,153 @@
+/*
+Copyright © 2023 VMware, Inc. All Rights Reserved.
+SPDX-License-Identifier: MPL-2.0
+*/
+
+package spec
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/helper"
+	gitrepositoryclustermodel "github.com/vmware/terraform-provider-tanzu-mission-control/internal/models/gitrepository/cluster"
+)
+
+var SpecSchema = &schema.Schema{
+	Type:        schema.TypeList,
+	Description: "Spec for the Repository.",
+	Required:    true,
+	MaxItems:    1,
+	Elem: &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			URLKey: {
+				Type:         schema.TypeString,
+				Description:  "URL of the git repository. Repository URL should begin with http, https, or ssh",
+				Required:     true,
+				ValidateFunc: validation.IsURLWithScheme([]string{"http", "https", "ssh"}),
+			},
+			secretRefKey: {
+				Type:        schema.TypeString,
+				Description: "Reference to the secret. Repository credential.",
+				Optional:    true,
+				Default:     "",
+			},
+			intervalKey: {
+				Type:        schema.TypeString,
+				Description: "Interval at which to check gitrepository for updates. This is the interval at which Tanzu Mission Control will attempt to reconcile changes in the repository to the cluster. A sync interval of 0 would result in no future syncs. If no value is entered, a default interval of 5 minutes will be applied as `5m`.",
+				Optional:    true,
+				Default:     "300s",
```

Comment:
can you mention it in minutes ?

Reply:
If we mention this in minutes, the backend converts it into seconds and we have an unwanted diff.

Also, the UI & CLI sends the default value as `300s` for this field. 

Let me know if there are any concerns with mentioning it in seconds instead of minutes.

Reply:
As discussed, I have added a `DiffSuppressFunc` for the intervals which will allow users to specify the same in minutes as well.

The default value has been updated to `5m` as part of this change.